File: frontend/lib/models/piece_metrage.dart
/// Une pièce dans une fiche de métrage (Module 5 — jusqu'à 30 pièces).
class PieceMetrage {
  final String nomPiece;
  final double longueur;
  final double largeur;
  final double surfaceDeduction; // poutres, piliers, gaines, baies

  PieceMetrage({
    required this.nomPiece,
    required this.longueur,
    required this.largeur,
    this.surfaceDeduction = 0,
  });

  double get surfaceBrute => longueur * largeur;
  double get surfaceNette => surfaceBrute - surfaceDeduction;
}

File: apps/api/src/main/java/com/eliteplaco/api/entity/PieceMetrage.java
package com.eliteplaco.api.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

/** Une pièce dans une fiche de métrage (longueur/largeur en mètres). */
@Entity
@Table(name = "piece_metrage")
@Getter
@Setter
public class PieceMetrage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "fiche_metrage_id", nullable = false)
    private FicheMetrage fiche;

    @Column(name = "nom_piece", nullable = false)
    private String nomPiece;

    @Positive
    @Column(nullable = false, precision = 6, scale = 2)
    private BigDecimal longueur;

    @Positive
    @Column(nullable = false, precision = 6, scale = 2)
    private BigDecimal largeur;

    @Column(name = "surface_deduction", nullable = false, precision = 6, scale = 2)
    private BigDecimal surfaceDeduction = BigDecimal.ZERO;

    private Integer ordre = 1;

    // Calculs dérivés — volontairement absents ici (voir MetrageService),
    // l'entité JPA ne porte aucune règle métier (même principe que Chantier).
}
